src/mono/mono/metadata/class-init.c

```diff
@@ -816,7 +816,7 @@ has_inline_array_attribute_value_func (MonoImage *image, uint32_t method_token,
 		MonoDecodeCustomAttr *decoded_attr = mono_reflection_create_custom_attr_data_args_noalloc (image, ctor, (guchar*)data, data_size, &error);
 		mono_error_assert_ok (&error);
 		g_assert (decoded_attr->named_args_num == 0 && decoded_attr->typed_args_num == 1);
-		attr->value = *(gpointer*)decoded_attr->typed_args [0]->value.primitive;
+		attr->value = (gpointer)*(guint32*)decoded_attr->typed_args [0]->value.primitive;
```

Comment:
@uweigand Sorry, I completely misunderstood the issue.  `load_cattr_value_noalloc` is already using `read16`/`read32`/`read64`.  The problem is that the value is read out as a `gpointer` even though `value.primitive` is really a `guint32*`.  Your fix is the right thing.

(minor quibble: cast from `guint32` to `gpointer` will elicit a warning on some picky compilers. So use `GUINT_TO_POINTER`)

```suggestion
		attr->value = GUINT_TO_POINTER(*(guint32*)decoded_attr->typed_args [0]->value.primitive);
```

Reply:
Thanks!  I saw the failures, and have now pushed an updated version.